control.Dockerfile

```diff
@@ -0,0 +1,70 @@
+# Build Stage
+################################################################################
+FROM rust:1.54-slim-buster as builder
+
+RUN rustup component add clippy
+
+RUN apt-get update \
+  && apt-get install -y ca-certificates pkg-config libssl-dev \
+  && rm -rf /var/lib/apt/lists/*
+
+# We need the SQLx tools to setup the test database.
+RUN cargo install sqlx-cli --version="0.5.10"
+
+ENV RUST_BACKTRACE=1
+
+WORKDIR /app
+
+COPY crates/control/Cargo.toml ./Cargo.lock ./
+
+# Avoid having to install/build all dependencies by copying the Cargo files and
+# making a dummy src/main.rs and empty lib.rs files.
+RUN mkdir -p ./src/bin \
```

Comment:
No action for now, but I just want to leave a note here about how we're building images because I'm noticing some inconsistencies in how we do it. Some binaries are built inside of a multistage docker build like this, and others are built outside of docker and copied into the container. I suppose you went with this route because otherwise you'd have no way to build an image locally due to the local glibc version being more recent than the one in the image. Makes me wonder if we should revisit having a flow-builder image so that we can have a consistent environment for the build stage.

Reply:
Yep, spot on. I can't build Flow images locally. This is a variation on the `source-kafka` Dockerfile, but I'm sure there's room for improvement. I'm no Docker wizard.

I went back and forth several times on whether we should be running the tests from within the Docker build here. Threading the network and postgres host feels weird, but it might just be a product of doing it for the first time. `source-postgres` uses the same trick. That said, I really didn't want to rebuild everything outside of Docker on the CI machine to avoid threading a build arg. I'm not sure there's a fantastic answer here. :shrug: